### PR TITLE
expose Spanner Options

### LINKED ISSUE #591: configurability of SpannerOptions and SessionPoolOptions as properties
Expose the options as properties available on [`SpannerOptions`](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java), it is very important to have a look at `SessionPoolOptions` in particular as large use cases will want fine tuneability on these knobs. 

Please discuss here first the details before jumping into implementation.

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
fixes #591

### CODE REVIEW

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java [hunk 1]
```diff
@@ -78,15 +94,59 @@ public GcpSpannerAutoConfiguration(GcpSpannerProperties gcpSpannerProperties,
...
 	@ConditionalOnMissingBean
-	public SpannerOptions spannerOptions() {
-		return SpannerOptions.newBuilder()
+	public SpannerOptions spannerOptions(SessionPoolOptions sessionPoolOptions) {
+		Builder builder = SpannerOptions.newBuilder()
 				.setProjectId(this.projectId)
 				.setHeaderProvider(new UsageTrackingHeaderProvider(this.getClass()))
-				.setCredentials(this.credentials).build();
+				.setCredentials(this.credentials);
+		if (this.numRpcChannels > 0) {
```
Comment (balopat) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java]: I think you meant `>=0` as the default is -1?

Comment (balopat) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java]: (this applies for the rest of the properties too)

Comment (ChengyuanZhao) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java]: fixed!

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java [hunk 2]
```diff
@@ -78,15 +94,59 @@ public GcpSpannerAutoConfiguration(GcpSpannerProperties gcpSpannerProperties,
...
 				.setHeaderProvider(new UsageTrackingHeaderProvider(this.getClass()))
-				.setCredentials(this.credentials).build();
+				.setCredentials(this.credentials);
+		if (this.numRpcChannels >= 0) {
+			builder.setNumChannels(this.numRpcChannels);
+		}
+		if (this.prefetchChunks >= 0) {
+			builder.setPrefetchChunks(this.prefetchChunks);
+		}
+		if (sessionPoolOptions != null) {
```
[the remark this answers is not in the document]
Comment (ChengyuanZhao) [reply spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerAutoConfiguration.java]: I will remove the check. My original thought is that  the user can override the sessionpooloptions bean with one of his own that does return null. (But in both cases the SpannerOptions  is ok with it being null)

**Pull request merged**